release.sh

```diff
@@ -0,0 +1,660 @@
+#			Function: check_local_changes
+#------------------------------------------------------------------------------
+#
+check_local_changes() {
+    git diff --quiet HEAD > /dev/null 2>&1
+    if [ $? -ne 0 ]; then
+	echo ""
+	echo "Uncommitted changes found. Did you forget to commit? Aborting."
+	echo ""
+	echo "You can perform a 'git stash' to save your local changes and"
+	echo "a 'git stash apply' to recover them after the tarball release."
+	echo "Make sure to rebuild and run 'make distcheck' again."
+	echo ""
+	echo "Alternatively, you can clone the module in another directory"
+	echo "and run ./configure. No need to build if testing was finished."
+	echo ""
+	return 1
+    fi
+    return 0
+}
+
+#------------------------------------------------------------------------------
+#			Function: check_option_args
+#------------------------------------------------------------------------------
+#
+# perform sanity checks on cmdline args which require arguments
+# arguments:
+#   $1 - the option being examined
+#   $2 - the argument to the option
+# returns:
+#   if it returns, everything is good
+#   otherwise it exit's
+check_option_args() {
+    option=$1
+    arg=$2
+
+    # check for an argument
+    if [ x"$arg" = x ]; then
+	echo ""
+	echo "Error: the '$option' option is missing its required argument."
+	echo ""
+	usage
+	exit 1
+    fi
+
+    # does the argument look like an option?
+    echo $arg | $GREP "^-" > /dev/null
+    if [ $? -eq 0 ]; then
+	echo ""
+	echo "Error: the argument '$arg' of option '$option' looks like an option itself."
+	echo ""
+	usage
+	exit 1
+    fi
+}
+
+#------------------------------------------------------------------------------
+#			Function: check_modules_specification
+#------------------------------------------------------------------------------
+#
+check_modules_specification() {
+
+if [ x"$MODFILE" = x ]; then
+    if [ x"${INPUT_MODULES}" = x ]; then
+	echo ""
+	echo "Error: no modules specified (blank command line)."
+	usage
+	exit 1
+    fi
+fi
+
+}
+
+#------------------------------------------------------------------------------
+#			Function: generate_announce
+#------------------------------------------------------------------------------
+#
+generate_announce()
+{
+    cat <<RELEASE
+Subject: [ANNOUNCE] $pkg_name $pkg_version
+To: $list_to
+Cc: $list_cc
+
+Nicknamed "my releaser forgot to named me".
+
+libratbag $tag_name is out.
+
+The list of interesting changes are:
+
+See the full log at the end if you are interested in the details.
+
+Cheers,
+Your releaser
+
+`git log --no-merges "$tag_range" | git shortlog`
+
+git tag: $tag_name
+
+RELEASE
+
+    for tarball in $tarbz2 $targz $tarxz; do
+	cat <<RELEASE
+As always, no tar.gz release on a different server, but you can grab
+the tag directly from github:
+
+https://github.com/libratbag/libratbag/archive/$tag_name/$tarball
+MD5:  `$MD5SUM $tarball`
+SHA1: `$SHA1SUM $tarball`
+SHA256: `$SHA256SUM $tarball`
```

Comment:
Sadly yes, they are timestamped. I kept around the test of downloading the link still, just in case something broke in the future :)